### PR TITLE
feat: add option to skip signature on macos

### PR DESCRIPTION
**Pull request by Mikescops**
There are scenarios where one would want to sign manually or later the binaries.

I'm adding a simple option that will skip the final codesign on macos.

### CODE REVIEW

**Review by robertsLando — APPROVED** [review]

**Comment by Mikescops** [comment]
@baparham whenever you have time to take a look here 🙌

**Comment by baparham** [comment]
Can you come up with a test for this? There seems to be one other signature test but I'm not entirely sure how you could replicate it to test this feature and only run on mac, but I welcome your thoughts, otherwise this is untested code (even though it is simple enough to 'test' in your brain)

https://github.com/vercel/pkg/blob/main/test/test-50-signature/main.js

**Comment by Mikescops** [comment]
@baparham I come up with a solution that works well on my mac, but it seems like there are no CI for macos-latest currently

**Comment by baparham** [comment]
Ah, of course. We should fix that, are you comfortable adding to the workflow matrix in this pr?

**Comment by baparham** [comment]
let's see! Thanks

**Comment by Mikescops** [comment]
Macos is damn too smart... 
image

**Comment by Mikescops** [comment]
@baparham ok I made some extra fixes and I could run all tests on my computer

File: test/test-50-reproducible/main.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ const output2 = './test-output-2.exe';
 
 utils.pkg.sync([
   '--public',
+  '--no-signature', // the signature will make the build not reproducible
```
Comment (baparham) [inline test/test-50-reproducible/main.js:23]: how did this work before? is this just applicable to running test-50-reproducible on macos now?

**Comment by Mikescops** [comment]
Sounds good ✅

**Comment by Mikescops** [comment]
I'd recommend dropping node12 support to lighten a bit the CI

**Comment by baparham** [comment]
> I'd recommend dropping node12 support to lighten a bit the CI

I 100% agree with this. I see no reason to support an EOL node js publicly with pkg.

I'm not sure what SLA we claim in pkg though, so this probably needs some more discussion from @robertsLando and maybe @leerob

**Comment by Mikescops** [comment]
I have opened a separate issue as we can move forward on this specific PR first https://github.com/vercel/pkg/issues/1886

Is it ok for you to merge?

**Review by baparham — APPROVED** [review]
lgtm

**Comment by Mikescops** [comment]
why this merge commit?

**Comment by baparham** [comment]
> why this merge commit?

It seems the repo is set up to require the branch be up to date before I can merge it. I don't really know why, but now it says it can be merged in, which I'll do in a squash commit. Perhaps the requirement is that the checks pass on the final merged result or something against `main` 🤷

**Comment by Mikescops** [comment]
Weird, I made a rebase on main before pushing 🤔

Thanks for the help here!

**Comment by Mikescops** [comment]
@baparham @robertsLando can we tag a release? i'd need to use the new option soon 👍

**Comment by robertsLando** [comment]
@Mikescops Unfortunately this doesn't depends by us as we both don't have the permissions to make new releases. @baparham already sent a private message to @leerob for this, I think it will be ready for the end of this week.

BTW in the meanwhile you could simply `npm install vercel/pkg#master`

**Comment by Mikescops** [comment]
Ok thanks, I'll wait then. I can't really do this in a production software unfortunately 😂

**Comment by robertsLando** [comment]
@Mikescops Giving that you just need to use it to package the application (it's a dev dependency) that will give the same result as after the release so not really a big difference

**Comment by Mikescops** [comment]
Not really, because it will be used in a brew Formula to package the app, and it is versioned, and I can't guarantee what will be added to the main branch here 'till i can update the brew Formula.

**Pull request merged**